.github/copilot-instructions.md

```diff
@@ -0,0 +1,255 @@
+
+### Initial Repository Setup
+
+1. **Clone and navigate to repository:**
+   ```bash
+   git clone https://github.com/dotnet/maui.git
+   cd maui
+   ```
+
+2. **Restore tools and build tasks (REQUIRED before opening IDE):**
+   ```bash
+   dotnet tool restore
+   dotnet build ./Microsoft.Maui.BuildTasks.slnf
+   ```
+
+## Project Structure
+
+### Important Directories
+- `src/Core/` - Core MAUI framework code
+- `src/Controls/` - UI controls and components
+- `src/Essentials/` - Platform APIs and essentials
+- `src/TestUtils/` - Testing utilities and infrastructure
+- `docs/` - Development documentation
+- `eng/` - Build engineering and tooling
+- `.github/` - GitHub workflows and configuration
+
+### Platform-Specific Code Organization
+- **Android** specific code is inside folders labeled `Android`
+- **iOS** specific code is inside folders labeled `iOS`
+- **MacCatalyst** specific code is inside folders named `MacCatalyst`
+- **Windows** specific code is inside folders named `Windows`
+
+### Solution Files
+- `Microsoft.Maui.sln` - Main solution file
+- `Microsoft.Maui-windows.slnf` - Windows-specific solution filter
+- `Microsoft.Maui-mac.slnf` - macOS-specific solution filter
+- `Microsoft.Maui-vscode.sln` - VS Code optimized solution
+- `Microsoft.Maui.BuildTasks.slnf` - Build tasks solution (must build first)
+
+### Sample Projects
+```
+├── Controls 
+│   ├── samples
+│   │   ├── Maui.Controls.Sample
+│   │   ├── Maui.Controls.Sample.Sandbox
+├── Essentials 
+│   ├── samples
+│   │   ├── Essentials.Sample
+├── BlazorWebView 
+│   ├── samples
+│   │   ├── BlazorWinFormsApp
+│   │   ├── BlazorWpfApp
+```
+
+- `src/Controls/samples/Maui.Controls.Sample` - Full gallery sample with all controls and features
+- `src/Controls/samples/Maui.Controls.Sample.Sandbox` - Empty project for testing/reproduction
+- `src/Essentials/samples/Essentials.Sample` - Essentials API demonstrations (non-UI MAUI APIs)
+- `src/BlazorWebView/samples/` - BlazorWebView sample applications
+
+## Development Workflow
+
+### IDE Setup
+
+#### VS Code (Recommended for Linux/macOS)
+1. Open repository root folder in VS Code
+2. Install the ".NET MAUI Dev Kit" extension from the marketplace: https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-maui
+3. Wait for IntelliSense to initialize (may take several minutes to fully process the solution)
+4. Use Command Palette (`Ctrl+Shift+P`):
+   - "Pick Device" to select target platform
+   - "Pick Startup Project" to select project to run
+
+**Note:** IntelliSense takes considerable time to fully process the solution. If experiencing issues, unload/reload the `maui.core` and `maui.controls` projects to resolve problems.
+
+#### Visual Studio (Windows)
+- Install Visual Studio 2022 v17.12 or newer
+- Follow [these steps](https://learn.microsoft.com/dotnet/maui/get-started/installation?tabs=vswin) to include MAUI workload
+- Open `Microsoft.Maui-windows.slnf` in Visual Studio from the repository root
```

Comment:
Maybe we remove `IDE Setup` completely, as the AI won't be doing this on its own.